### DISCUSSION TITLE
Allow clipboard paste to terminal

### DISCUSSION THREAD

**Discussion by mitchplze** [louislam/dockge#141] [Feature Request]
Currently I am unable to paste clipboard contents (commands) to the Dockge terminal in Safari 17.1 on MacOS.

**Comment by cirrusflyer** [comment]
Same here. Definitely need this ability.

**Comment by cirrusflyer** [comment]
Any update on this?

**Comment by vijay-prema** [comment]
Same in Firefox 122. Ubuntu 2204.

**Comment by arpanghosh8453** [comment]
I am looking forward to this too!

**Comment by RandomLegend** [comment]
Another updoot from my side aswell!

**Comment by lobovkin** [comment]
xtermjs (which is used for the therminal here) has this feature. Maybe, you just need to update the dependency.

**Reply by explorer369** [reply]
How do I do this?

**Reply by ymickler** [reply]
> How do I do this?

Looks like you need to add the xterm adddon: https://github.com/xtermjs/xterm.js/tree/master/addons/addon-clipboard

**Comment by ghost** [comment]
Same, chromium (Brave browser and Edge) Windows 11

**Comment by itsbaraa** [comment]
I need this feature.

**Reply by ghost** [reply]
> I need this feature.

We all do lol. Has the owner of this project abandoned it or something?

**Reply by itsbaraa** [reply]
I think he's focusing on uptime kuma at the moment, for me uptime kuma is working perfectly and doesn't need any more improvements, but dockge can be improved a lot

**Comment by vijay-prema** [comment]
There is a slightly janky but reliable [workaround](https://askubuntu.com/a/218306/1064531) for this issue (actually works in any app lacking clipboard paste support).  It is specific to Linux host with x11.

TLDR; run `sh -c 'sleep 2; xdotool type "$(xclip -o -selection clipboard)"'` in the host **local terminal** then you have 2 seconds to click in the Dockge terminal and it will "paste" _by typing out the clipboard contents_.  Adjust the sleep time if you need to.  Better yet, bind it to a hotkey.

**Reply by hg42** [reply]
thx for the idea, usually I would also think of such a command, but I didn't for some unknown reason

**Comment by kbsmidt** [comment]
Copy/paste functionality in the Dockge terminal will definitely elevate Dockge to new heights.

**Reply by InnocentRain** [reply]
I absolutly agree, this is a massive QOL improvement :)

**Comment by SIlver--** [comment]
Looks like it's in the pull requests, so should hopefully be accepted soon

https://github.com/louislam/dockge/pull/623

**Reply by ghost** [reply]
Soon tm

**Comment by naujramirez** [comment]
Couldn't get xclip and xdotool to work, but they led me to this [script](https://gist.github.com/ethack/110f7f46272447828352768e6cd1c4cb).

This is what worked for me. It allowed me to tell the mac os terminal to delay the execution of the command that simulates typing text from your clipboard. 

- Command that simulates typing text from clipboard:
osascript -e 'tell application "System Events" to keystroke the clipboard as text'

- Command that delays execution of a command by 3 seconds:
sleep 3 &&  "place command here"

- Together:
sleep 3 && osascript -e 'tell application "System Events" to keystroke the clipboard as text'

**Comment by rickdgray** [comment]
would love this feature

**Comment by pogora** [comment]
Agreed that would be a massive QOL improvement.  Currently that's why I am still using Portainer as well, since it works there. 
Although I'd like to only use Dockge.

**Comment by myths420** [comment]
any update on this? isnt this a common sense feature that should of been working on day one. over a year and nothing.

**Comment by nigiriemoji** [comment]
When is this happening?

**Reply by rickdgray-dc** [reply]
Louis isn't getting back to this for a while
https://www.reddit.com/r/selfhosted/comments/1hee1h6/comment/m23onzs/

**Comment by artamrj** [comment]
@louislam Still no update?

**Comment by LvNaTiCC** [comment]
This feature would really be appreciated

**Comment by 14790897** [comment]
It's very uneasy to manually enter command in this AI era.

**Comment by realindiahotel** [comment]
Everyone this is fixed in the nightly build. Switch from latest to nightly and it is working!
